src/test/regress/sql/multi_hash_pruning.sql

```diff
@@ -27,11 +27,40 @@ SET client_min_messages TO DEBUG2;
 -- Check that we can prune shards for simple cases, boolean expressions and
 -- immutable functions.
 
+
+-- Since router plans are not triggered for task-tracker executor type,
+-- we need to run the tests that triggers router planning seperately for
+-- both executors. Otherwise, check-full fails on the task-tracker.
+-- Later, we need to switch back to the actual task executor
+-- to contuinue with correct executor type for check-full.
+SELECT quote_literal(setting) AS actual_task_executor
```

Comment:
Pruning mainly concerns planning, not execution, but ok, if you switch to using current_setting this is 🚢